precompiles/wrapper.go

```diff
@@ -117,3 +117,67 @@ func (wrapper *OwnerPrecompile) Precompile() *Precompile {
 	con := wrapper.precompile
 	return con.Precompile()
 }
+
+// NativeTokenPrecompile is a precompile wrapper for those only native token owners may use
+type NativeTokenPrecompile struct {
```

Comment:
I don't like duplicating this bunch of code that is already a little scary and not well understood.
Better and easier to review adding specific logic to the mint/burn functions that checks the caller.

See ArbWasmCache.go, and how it implements and uses "hasAccess" (in our case we shouldn't allow chain owner access)

Reply:
Refactored.